src/TheFox/FlickrCli/Command/PiwigoCommand.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace TheFox\FlickrCli\Command;
+
+use Doctrine\DBAL\Configuration;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DriverManager;
+use Exception;
+use Rezzza\Flickr\ApiFactory;
+use Rezzza\Flickr\Http\GuzzleAdapter;
+use Rezzza\Flickr\Metadata;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Yaml\Yaml;
+
+class PiwigoCommand extends Command
+{
+
+    /** @var string[] Array of photoset titles, keyed by their ID. */
+    protected $photosets;
+
+    protected function configure()
+    {
+        $this->setName('piwigo');
+        $this->setDescription('Upload files from Piwigo to Flickr');
+
+        $this->addOption('config', 'c', InputOption::VALUE_OPTIONAL, 'Path to config file; default: config.yml');
+        $this->addOption('log', 'l', InputOption::VALUE_OPTIONAL, 'Path to log directory; default: log');
+
+        $this->addOption('dbname', null, InputOption::VALUE_REQUIRED, 'Name of the Piwigo database');
+        $this->addOption('dbuser', null, InputOption::VALUE_REQUIRED, 'User as whom to connect to the Piwigo database');
+        $this->addOption('dbpass', null, InputOption::VALUE_REQUIRED, 'Password for the Piwigo database user');
+        $this->addOption('dbhost', null, InputOption::VALUE_REQUIRED, 'The Piwigo database host name', 'localhost');
+
+        $this->addOption('piwigo-uploads', null, InputOption::VALUE_REQUIRED, "Path to the Piwigo 'uploads' directory");
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        // Piwigo.
+        $dbConfig = new Configuration();
+        $connectionParams = array(
+            'dbname' => $input->getOption('dbname'),
```

Comment:
Maybe you can move the Piwigo settings to `config.yml`, so we don't have separate options here.

Like this:

```yaml
flickr:
...
piwigo:
  dbname: ...
  user: ...
  password: ...
  host: ...
  driver: pdo_mysql

Reply:
Good idea. Done.